conda_forge_tick/migrators.xsh

```diff
@@ -368,3 +368,59 @@ class Compiler(Migrator):
 
     def remote_branch(self):
         return 'compiler_migration2'
+
+
+class Noarch(Migrator):
+    """Migrator for adding noarch."""
+    migrator_version = 0
+
+    compiler_pat = re.compile('.*_compiler_stub')
+    sel_pat = re.compile('(.+?)\s*(#.*)?\[([^\[\]]+)\](?(2)[^\(\)]*)$')
+    python_pat = re.compile('python( )?(?(1).*)')
+    unallowed_reqs = ['toolchain', 'gcc', 'cython', 'clangdev']
+
+    def filter(self, attrs):
```

Comment:
2to3 lives here https://github.com/PyGithub/PyGithub/blob/master/setup.py#L104

Reply:
from https://github.com/conda-forge/pygithub-feedstock/pull/21

Reply:
(maybe we need to capture the `setup.py` somewhere?)

Reply:
That can be very complex and error prone. I would only add warnings for `2to3` and other hacks in `setup.py` that may break `noarch`.

Reply:
Hmm ok, although I think down the line we should think about source inspection (esp. since this was a goal for libcflib) (eg dep finding).